sdk/storage/storage-file-datalake/review/storage-file-datalake.api.md

```diff
@@ -0,0 +1,1424 @@
+    // (undocumented)
+    group?: string;
+    // (undocumented)
+    owner?: string;
+}
+
+// Warning: (ae-forgotten-export) The symbol "PathSetAccessControlHeaders" needs to be exported by the entry point index.d.ts
+//
+// @public
+type PathSetAccessControlResponse = PathSetAccessControlHeaders & {
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: PathSetAccessControlHeaders;
+    };
+};
+
+export { PathSetAccessControlResponse }
+
+export { PathSetAccessControlResponse as PathSetPermissionsResponse }
+
+// @public (undocumented)
+export interface PathSetHttpHeadersHeaders {
+    // (undocumented)
+    clientRequestId?: string;
+    // (undocumented)
+    date?: Date;
+    // (undocumented)
+    etag?: string;
+    // (undocumented)
+    lastModified?: Date;
+    // (undocumented)
+    requestId?: string;
+    // (undocumented)
+    version?: string;
+}
+
+// @public (undocumented)
+export interface PathSetHttpHeadersOptions extends CommonOptions {
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: DataLakeRequestConditions;
+}
+
+// @public (undocumented)
+export type PathSetHttpHeadersResponse = PathSetHttpHeadersHeaders & {
+    _response: HttpResponse & {
+        parsedHeaders: PathSetHttpHeadersHeaders;
+    };
+};
+
+// @public (undocumented)
+export interface PathSetMetadataHeaders {
+    // (undocumented)
+    clientRequestId?: string;
+    // (undocumented)
+    date?: Date;
+    // (undocumented)
+    encryptionKeySha256?: string;
+    // (undocumented)
+    etag?: string;
+    // (undocumented)
+    isServerEncrypted?: boolean;
+    // (undocumented)
+    lastModified?: Date;
+    // (undocumented)
+    requestId?: string;
+    // (undocumented)
+    version?: string;
+}
+
+// @public (undocumented)
+export interface PathSetMetadataOptions extends CommonOptions {
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: DataLakeRequestConditions;
+}
+
+// @public (undocumented)
+export type PathSetMetadataResponse = PathSetMetadataHeaders & {
+    _response: HttpResponse & {
+        parsedHeaders: PathSetMetadataHeaders;
+    };
+};
+
+// @public (undocumented)
+export interface PathSetPermissionsOptions extends CommonOptions {
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: DataLakeRequestConditions;
+    // (undocumented)
+    group?: string;
+    // (undocumented)
+    owner?: string;
+}
+
+// Warning: (ae-forgotten-export) The symbol "PathUpdateHeaders" needs to be exported by the entry point index.d.ts
+//
+// @public
+type PathUpdateResponse = PathUpdateHeaders & {
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: PathUpdateHeaders;
+    };
+};
+
+export { PathUpdateResponse as FileAppendResponse }
+
+export { PathUpdateResponse as FileFlushResponse }
+
+// @public
+export class Pipeline extends Pipeline_2 {
```

Comment:
Why do we extend storage-blob's pipeline? Can we either just use storage-blobs version (since we have the dependency now anyway), or else not extend?

Reply:
@beterlson It's a workaround because of the 2 `Pipeline` class definitions in datalake and blob package. `instanceof Pipeline` isused in BlobXXXClient constructors. When initializing internal BlobXXXClient objects, datalake package needs a blob package Pipeline instance. 

The most straightforward way is to create Pipeline converter. Convert datalake Pipeline to blob Pipeline. But it needs to convert a whole set of objects from datalake version to blob version, like credentials, itera through policy factory array and convert policies instances.

We can also removing `instanceof Pipeline` from BlobXXXClient constructors, but I don't have a better replacement.

Comparing with above 2 choices, extends from blob Pipeline is a clean & simple workaround.

As you mentioned, we can also reuse blob Pipeline as datalake pipeline. This comes to a question. Should we reuse much of models, classess, interfaces from blob package, or create new. In previous .Net/Java API review, the consulusion is we should create and maintain new set of models/classes for datalake package. It makes sense because datalake customers don't like models hide deep in another package, and they don't need learn concepts from blob package. It gives future flexibility too.

Reply:
Agree in general about creating new types. I see the constraints you're working under now, thanks for explaining! I'm ok with the current approach, but one comment: 

> We can also removing instanceof Pipeline from BlobXXXClient constructors, but I don't have a better replacement.

I don't think this check is providing much value, maybe the opposite. Instanceof is so brittle. Better to check for existence of properties you need, or even do no checking at all I think.